pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/AbstractIndexType.java

```diff
@@ -51,7 +51,11 @@ public Map<String, C> getConfig(TableConfig tableConfig, Schema schema) {
     if (_deserializer == null) {
       _deserializer = createDeserializer();
     }
-    return _deserializer.deserialize(tableConfig, schema);
+    try {
+      return _deserializer.deserialize(tableConfig, schema);
```

Comment:
Can you help me understand why the original code doesn't have to deal with exception?

Reply:
With which exception? `ConfigDeclaredTwiceException`? Here we are dealing with that exception. But it wraps it in order to also inform about which index type is the one that is failing. The code that throws the original exception doesn't have the context to know which index type is the one that is repeated.